### PR TITLE
8.0 l10n br account payment mode

### PR DESCRIPTION
**Pull request by rvalyi**

### CODE REVIEW

File: l10n_br_account_payment_mode/README.rst [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+Installation
+============
+
+This module depends on:
+
+* account_banking_payment_export
+* l10n_br_account
+* l10n_br_data_base
+* account_due_list_payment_mode
+* account_banking_payment_export
```
Comment (rvalyi) [inline l10n_br_account_payment_mode/README.rst]: account_banking_payment_export dependency is listed twice

File: l10n_br_account_payment_mode/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    'license': 'AGPL-3',
+    'author': "KMEE, Odoo Community Association (OCA)",
+    "website": "https://odoo-community.org/",
+    'depends': [
+        'l10n_br_data_base',
+        'l10n_br_account_product',
+        'account_due_list_payment_mode',
+        'account_banking_payment_export',
+        'l10n_br_account_banking_payment',
+    ],
```
Comment (rvalyi) [inline l10n_br_account_payment_mode/__openerp__.py]: these dependencies look fine but require OCA/bank-payment and OCA/account-payment to be added in the oca_dependencies.txt file of the repo.
Also this PR depends on the merge of l10n_br_account_banking_payment in this repo. I tried to extract it with its relative commits but didn't succeed yet. I'll try it again soon.

File: l10n_br_account_payment_mode/report/__init__.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
```
Comment (rvalyi) [inline l10n_br_account_payment_mode/report/__init__.py]: prefer shorter license headers, specially for the __init__.py files

File: l10n_br_account_payment_mode/models/res_partner_bank.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+
+class ResPartnerBank(models.Model):
+    _inherit = 'res.partner.bank'
+
+    @api.multi
+    @api.constrains('bank_bic')
+    def check_bic_length(self):
+        '''
+        sobrescrever constrains do core que não leva em consideração bancos
```
Comment (rvalyi) [inline l10n_br_account_payment_mode/models/res_partner_bank.py]: nao sou eu o brasileiro mas o portugues parece zoado. Nao seria melhor 
"sobreescreve constraints do core..." ?

File: l10n_br_account_payment_mode/models/res_bank.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    _inherit = 'res.bank'
+
+    @api.multi
+    @api.constrains('bic')
+    def check_bic_length(self):
+        '''
+        sobrescrever constrains do core que nao leva em consideração bancos
+         que nao são intenacionais.
+        '''
+        return True
```
Comment (rvalyi) [inline l10n_br_account_payment_mode/models/res_bank.py]: nao sou eu o brasileiro mas o portugues parece zoado. Nao seria melhor
"sobreescreve constraints do core..." ?

File: l10n_br_account_payment_mode/models/payment_mode.py [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
```
Comment (rvalyi) [inline l10n_br_account_payment_mode/models/payment_mode.py]: short style license header please

File: l10n_br_account_payment_mode/__init__.py [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
```
Comment (rvalyi) [inline l10n_br_account_payment_mode/__init__.py]: short style license headers please

**Review by rvalyi — COMMENTED** [review]
as for tests, there is no much logic here, so just having the module installing is probably okay. Installation depends on the l10n_br_account_banking_payment merge though. Meanwhile minor changes are required. cc @mbcosta @mileo @renatonlima

**Comment by mbcosta** [comment]
Hi @rvalyi 
     Requested changes done , now just waiting review and merge of PR https://github.com/OCA/l10n-brazil/pull/586

File: l10n_br_account_payment_mode/README.rst [hunk 8]
```diff
@@ -0,0 +1,70 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License: AGPL-3
+
+Brazil Account Banking - Payments Mode
+======================================
+
+This module provide an mode payment for payment orders.
```
Comment (rvalyi) [inline l10n_br_account_payment_mode/README.rst]: s/provide/provides
s/an mode for payment orders/a mode for payment orders in Brazil

File: l10n_br_account_payment_mode/__init__.py [hunk 9]
```diff
@@ -0,0 +1,7 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2015-Today - KMEE (<http://kmee.com.br>).
+#  @author Luis Felipe Miléo - mileo@kmee.com.br
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from . import models
+
```
Comment (rvalyi) [inline l10n_br_account_payment_mode/__init__.py]: remove last line

File: l10n_br_account_payment_mode/models/__init__.py [hunk 10]
```diff
@@ -0,0 +1,8 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2017-Today - KMEE (<http://kmee.com.br>).
+#  @author Luis Felipe Miléo - mileo@kmee.com.br
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from . import payment_mode
+from . import account_invoice
+
```
Comment (rvalyi) [inline l10n_br_account_payment_mode/models/__init__.py]: remove last line

**Review by rvalyi — COMMENTED** [review]
minor changes requested

**Comment by mbcosta** [comment]
Changes done @rvalyi

**Comment by mbcosta** [comment]
Error on travis seems not be related with this PR

" no such option: --allow-external

The command "pip install https://github.com/aricaldeira/PySPED/archive/master.zip#egg=PySPED-1.1.0 --allow-external PyXMLSec --allow-insecure PyXMLSec" failed and exited with 2 during . "

**Comment by mbcosta** [comment]
Waiting merge https://github.com/OCA/l10n-brazil/pull/606

File: l10n_br_account_payment_mode/models/__init__.py [hunk 11]
```diff
@@ -0,0 +1,8 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2017-Today - KMEE (<http://kmee.com.br>).
+#  @author Luis Felipe Miléo - mileo@kmee.com.br
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from . import payment_mode
+from . import account_invoice
```
Comment (sadamo) [inline l10n_br_account_payment_mode/models/__init__.py]: There is no account_invoice file

**Review by mileo — APPROVED** [review]

**Review by sadamo — APPROVED** [review]

**Review by hendrixcosta — APPROVED** [review]

**Pull request merged**